client/app/bundles/course/material/folders/pages/FolderShow/index.tsx

```diff
@@ -0,0 +1,173 @@
+import EditFolderButton from '../../components/buttons/EditFolderButton';
+import MaterialUpload from '../../components/misc/MaterialUpload';
+
+import FolderNew from '../FolderNew';
+import FolderEdit from '../FolderEdit';
+
+interface Props extends WrappedComponentProps {}
+
+const translations = defineMessages({
+  defaultHeader: {
+    id: 'course.materials.folders.defaultHeader',
+    defaultMessage: 'Materials',
+  },
+});
+
+const FolderShow: FC<Props> = (props) => {
+  const { intl } = props;
+  const { folderId } = useParams();
+  const dispatch = useDispatch<AppDispatch>();
+
+  // For new folder form dialog
+  const [isNewFolderOpen, setIsNewFolderOpen] = useState(false);
+  // For edit folder form dialog
+  const [isEditFolderOpen, setIsEditFolderOpen] = useState(false);
+  // For material upload form dialog
+  const [isMaterialUploadOpen, setIsMaterialUploadOpen] = useState(false);
+
+  const id = useSelector((state: AppState) => getFolderId(state));
+  const parentId = useSelector((state: AppState) => getFolderParentId(state));
+  const name = useSelector((state: AppState) => getFolderName(state));
+  const description = useSelector((state: AppState) =>
+    getFolderDescription(state),
+  );
+  const canStudentUpload = useSelector((state: AppState) =>
+    getFolderCanStudentUpload(state),
+  );
+  const startAt = useSelector((state: AppState) => getFolderStartAt(state));
+  const endAt = useSelector((state: AppState) => getFolderEndAt(state));
+  const subfolders = useSelector((state: AppState) =>
+    getFolderSubfolders(state),
+  );
+  const materials = useSelector((state: AppState) => getFolderMaterials(state));
+  const permissions = useSelector((state: AppState) =>
+    getFolderPermissions(state),
+  );
+
+  const [isLoading, setIsLoading] = useState(true);
+  useEffect(() => {
+    if (folderId) {
+      dispatch(loadFolder(+folderId)).finally(() => setIsLoading(false));
+    }
+  }, [dispatch, folderId]);
+
+  if (isLoading) {
+    return <LoadingIndicator />;
+  }
+
+  const headerToolbars: ReactElement[] = [];
+
+  if (permissions.isConcrete && permissions.canCreateSubfolder) {
+    headerToolbars.push(
+      <NewSubfolderButton
+        key="new-folder-button"
+        handleOnClick={(): void => {
+          setIsNewFolderOpen(true);
+        }}
+      />,
+    );
+  }
+  if (permissions.isConcrete && permissions.canUpload) {
+    headerToolbars.push(
+      <UploadFilesButton
+        key="upload-files-button"
+        handleOnClick={(): void => {
+          setIsMaterialUploadOpen(true);
+        }}
+      />,
+    );
+  }
+  headerToolbars.push(
+    <DownloadFolderButton key="download-folder-button" currFolderId={id} />,
+  );
+  if (permissions.isConcrete && permissions.canEdit) {
+    headerToolbars.push(
+      <EditFolderButton
+        key="edit-folder-button"
+        handleOnClick={(): void => setIsEditFolderOpen(true)}
+      />,
+    );
+  }
+
+  const folderInitialValues = {
```

Comment:
FolderEdit is used in FolderShow index and the edit button for subfolders as well. So the initial values need to be declared higher and passed down since they are all different